app/src/main/java/org/kiwix/kiwixmobile/zim_manager/ZimManageViewModel.kt

```diff
@@ -108,6 +108,9 @@ class ZimManageViewModel @Inject constructor(
   val fileSelectActions = PublishProcessor.create<FileSelectActions>()
   val requestDownloadLibrary = BehaviorProcessor.createDefault<Unit>(Unit)
   val requestFiltering = BehaviorProcessor.createDefault<String>("")
+  val currentPage = PublishProcessor.create<Int>()
+
+  val libraryFragmentIsVisible = currentPage.filter { it == 1 }
```

Comment:
Great work, but now this is in a ViewModel so I would request unit tests for this filter.

if you look at ZimManageViewModelTest
```
    @Test
    fun `RestartActionMode offers StartMultiSelection`() {
      viewModel.sideEffects.test()
        .also { viewModel.fileSelectActions.offer(RestartActionMode) }
        .assertValues(StartMultiSelection(viewModel.fileSelectActions))
    }
```
even though it is a livedata the apis are very similar for testing disposables so your tests should bare a resemblance to this. Have 1 test where the filter emits and another where it doesn't, please.

Also on a fresh look I no longer like the name so `libraryTabIsVisible` is a teensy bit better I think, no need to leak an implementation detail in a name

Reply:
@macgills I'm facing a bit of difficulty over here. 
I offered the values 1 and 0 to the publishProcessor, `currentPage`. 1 for the case when user shifts to the online fragment and action mode has to be finished.
0 for when nothing has to be done. 
After this how do I check the filter? The disposable becomes a Flowable<Int!>

Reply:
you want to call `.test()` on the Flowable, under the hood this calls `.subscribe` with a `TestSubscriber`

as I said very similar to this

```
viewModel.sideEffects.test()
        .also { viewModel.fileSelectActions.offer(RestartActionMode) }
        .assertValues(StartMultiSelection(viewModel.fileSelectActions))
```

but `viewModel.sideEffects` is `libraryFragmentIsVisible `

Reply:
Thanks for this. I'll try to get this working ASAP.